universal-login-wallet/src/core/presenters/WalletPresenter.ts

```diff
@@ -5,6 +5,16 @@ export class WalletPresenter {
     private walletService: WalletService
   ) {}
 
+  getPrivateKey() {
+    switch (this.walletService.state.kind) {
+      case 'Connecting':
+      case 'Deployed':
+        return this.walletService.state.wallet.privateKey;
+      default:
+        throw new Error('Invalid wallet state: expected Deployed or Connecting wallet');
+    }
+  }
+
```

Comment:
Not sure about it. Don't know what is the difference between 'Connecting' and 'Deployed'. Made similarly to getName()

Reply:
That is a bit unintuitive indeed. It is fine for now.

In the future, we should sort out WalletService states better.
Perhaps `{Future, Deployed, Connected, Deploying, Connecting}`